src/components/UploadFile.tsx

```diff
@@ -0,0 +1,162 @@
+    fileLimit?: number;
+
+    /** The total size limit of the files that can be selected. */
+    totalFilesSizeLimit?: number;
+};
+
+function UploadFile({
+    buttonText,
+    uploadedFiles,
+    onUpload,
+    onRemove,
+    acceptedFileTypes,
+    style,
+    errorText = '',
+    setError,
+    onInputChange = () => {},
+    totalFilesSizeLimit = 0,
+    fileLimit = 0,
+}: UploadFileProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const theme = useTheme();
+
+    const handleFileUpload = (files: FileObject[]) => {
+        const totalSize = files.reduce((sum, file) => sum + (file.size ?? 0), 0);
+
+        if (totalFilesSizeLimit) {
+            if (totalSize > totalFilesSizeLimit) {
+                setError(translate('attachmentPicker.sizeExceededWithValue', {maxUploadSizeInMB: totalFilesSizeLimit / (1024 * 1024)}));
+                return;
+            }
+        }
+
+        if (fileLimit) {
+            if (files.length > 0) {
+                if (files.length > fileLimit) {
+                    setError(translate('attachmentPicker.tooManyFiles', {fileLimit}));
+                    return;
+                }
+            }
+        }
+
+        if (acceptedFileTypes.length > 0) {
+            const filesExtensions = files.map((file) => FileUtils.splitExtensionFromFileName(file?.name ?? '').fileExtension.toLowerCase());
+
+            if (acceptedFileTypes.every((element) => !filesExtensions.includes(element as string))) {
+                setError(translate('attachmentPicker.notAllowedExtension'));
+                return;
+            }
+        }
+
+        onInputChange(files);
+        onUpload(files);
+    };
+
+    return (
+        <View style={[styles.alignItemsStart, style]}>
+            <AttachmentPicker
+                acceptedFileTypes={acceptedFileTypes}
+                fileLimit={fileLimit}
+                allowMultiple={!!fileLimit}
+            >
+                {({openPicker}) => (
+                    <Button
+                        medium
+                        text={buttonText}
+                        accessibilityLabel={buttonText}
+                        onPress={() => {
+                            openPicker({
+                                onPicked: handleFileUpload,
+                            });
+                        }}
+                    />
+                )}
+            </AttachmentPicker>
+            {uploadedFiles.map((file) => (
+                <View
+                    style={[styles.flexRow, styles.alignItemsCenter, styles.justifyContentCenter, styles.border, styles.p4, styles.mt3]}
+                    key={file.uri}
+                >
+                    <Icon
+                        height={variables.iconSizeNormal}
+                        width={variables.iconSizeSemiSmall}
+                        src={Expensicons.Paperclip}
+                        fill={theme.textSupporting}
```

Comment:
Should be icon color, not textSupporting

Reply:
updated